src/lib/targeting.test.ts

```diff
@@ -1,192 +1,71 @@
+const withNowAs = <T>(now: Date, fn: () => T): T => {
+    const old = Date.now;
+    Date.now = () => now.valueOf(); // override
+    const got = fn();
+    Date.now = old;
+
+    return got;
+};
+
+describe('isRecentOneOffContributor', () => {
+    const now = new Date('2020-02-12T10:24:00');
+
+    it('returns true for recent date', () => {
+        const got = isRecentOneOffContributor(new Date('2020-02-10T10:24:00'), now);
+        expect(got).toBe(true);
     });
 
-    it('should return FALSE if content type IS NOT Article', () => {
-        const contentType = 'Feature';
-        expect(isEpicContent({ contentType })).toEqual(false);
+    it('returns false for older date', () => {
+        const got = isRecentOneOffContributor(new Date('2019-02-10T10:24:00'), now);
+        expect(got).toBe(false);
     });
-});
-
-describe('isEpicSuitable()', () => {
-    it('should return TRUE if all parameters ARE FALSE', () => {
-        const testSuitableFlags = {
-            shouldHideReaderRevenue: false,
-            isMinuteArticle: false,
-            isPaidContent: false,
-        };
 
-        expect(isEpicSuitable(testSuitableFlags)).toEqual(true);
-    });
-
-    it('should return FALSE if one or more parameters ARE TRUE', () => {
-        const testSuitableFlags1 = {
-            shouldHideReaderRevenue: true,
-            isMinuteArticle: false,
-            isPaidContent: false,
-        };
-        expect(isEpicSuitable(testSuitableFlags1)).toEqual(false);
-
-        const testSuitableFlags2 = {
-            shouldHideReaderRevenue: true,
-            isMinuteArticle: true,
-            isPaidContent: false,
-        };
-        expect(isEpicSuitable(testSuitableFlags2)).toEqual(false);
+    it('returns false for someone that has never contributed', () => {
+        const got = isRecentOneOffContributor(undefined, now);
+        expect(got).toBe(false);
     });
 });
 
-describe('isEpicWorthwhile()', () => {
-    it('should return TRUE if the section IS NOT blacklisted', () => {
-        const testSectionWorthwhile = {
-            sectionName: 'culture',
-            tags: [],
-        };
-
-        expect(isEpicWorthwhile(testSectionWorthwhile)).toEqual(true);
-    });
+describe('shouldNotRenderEpic', () => {
+    const meta: EpicTargeting = testData.targeting;
 
-    it('should return FALSE if the section IS blacklisted', () => {
-        const testSectionWorthwhile = {
-            sectionName: 'football',
-            tags: [],
-        };
+    const test = (m: EpicTargeting, want: boolean, now?: Date) => {
+        const nowDate = now ? now : new Date();
+        const got = withNowAs(nowDate, () => shouldNotRenderEpic(m));
+        expect(got).toBe(want);
+    };
 
-        expect(isEpicWorthwhile(testSectionWorthwhile)).toEqual(false);
+    it('returns true for non-article', () => {
+        const data = { ...meta, contentType: 'Liveblog' };
+        test(data, true);
```

Comment:
What do you think about instead of having the intermediate `test` method we use `withNowAs` directly when we need it? I think of the current invocations of `test` only one of them makes use of overriding the time, we otherwise default to `new Date()`. So I think it'd be nice to surface that intent in the test directly. Otherwise `test` becomes a wrapper for calling the method and making the assertion, which doesn't feel to me like it's quite pulling its weight (I think the duplication it removes is outweighed by the mental overload it takes to follow through to the method and find out what it does). What do you reckon? Particularly in tests I don't mind a little duplication if it makes them easier to read. Happy to pair or talk through IRL if it'd be helpful?!

Reply:
Yeah I agree, I ummed + aahed about this but will switch back!